src/instructlab/config/init.py

```diff
@@ -183,6 +195,48 @@ def init(
     )
 
 
+def hw_auto_detect() -> tuple[str | None, int | None, int | None, _train]:
+    profiles = render_profiles()
+    gpus = 0
+    total_vram = 0
+    train = _train()
+    chosen_vram = None
+    chosen_profile_name = None
+    previous_vram = None
+    # try nvidia
+    if torch.cuda.is_available():
+        click.echo("Detecting Hardware...")
+        for i in range(torch.cuda.device_count()):
+            properties = torch.cuda.get_device_properties(i)
+            total_vram += properties.total_memory  # memory in B
+    vram = convert_bytes_to_proper_mag(total_vram)[0]
+    for profile in profiles:
+        if vram < profile[0]:
+            for profile_names_and_values in profile[1]:
+                if train is not None:
+                    break
+                for i, vram_and_profile in enumerate(
+                    list(profile_names_and_values.values())
+                ):
+                    if vram < vram_and_profile[0]:
+                        chosen_profile_name = list(profile_names_and_values.keys())[i]
+                        chosen_vram = vram_and_profile[0]
```

Comment:
Is this format standardized in some way or can we use a dict with named keys here?  It feels a bit cryptic to follow.

Reply:
added comments why its this way, you can't put `_train` into a dict apparently, must be a tuple.

Reply:
I could be wrong here, but that is why I did it initially bc python told me the type was un-mappable if I recall correctly

Reply:
Were you using _train as the key?  That wouldn't be allowed because it is mutable.  I am not sure of anything that would keep _train being used as a value.  Would be good to know the specific error if that was the case.